jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -174,6 +180,8 @@ public QueuedThreadPool(@Name("maxThreads") int maxThreads, @Name("minThreads")
             int capacity = Math.max(_minThreads, 8) * 1024;
             queue = new BlockingArrayQueue<>(capacity, capacity);
         }
+        if (queue.remainingCapacity() != Integer.MAX_VALUE)
+            LOG.warn("A bounded thread pool queue may lead to unexpected behavior, use an unbounded queue instead.");
```

Comment:
```suggestion
            LOG.warn("Detected a bounded thread pool queue ({}), this can lead to unexpected behavior, use an unbounded queue instead.", TypeUtil.toShortName(queue.getClass()));
```